cpp/src/arrow/util/key_value_metadata.h

```diff
@@ -58,6 +59,12 @@ class ARROW_EXPORT KeyValueMetadata {
   ARROW_DISALLOW_COPY_AND_ASSIGN(KeyValueMetadata);
 };
 
+/// \brief Create a KeyValueMetadata instance
+///
+/// \param pairs key-value mapping
+std::shared_ptr<KeyValueMetadata> ARROW_EXPORT
+metadata(const std::unordered_map<std::string, std::string>& pairs);
```

Comment:
Since there are many kinds of "metadata" concepts in Arrow, we might want to call this `key_value_metadata` (or similar) to make it more clear

Reply:
Ok, renaming them